code/game/objects/items/tools/misc_tools.dm

```diff
@@ -7,7 +7,11 @@
 
 /atom/proc/set_name_label(var/new_label)
 	name_label = new_label
-	name = initial(name)
+	if (istype(src, /obj/item/reagent_container/food/snacks))
```

Comment:
istype(src) is awful from an OOP perspective, you should make this a thing on `/obj/item/reagent_container/food/snacks/set_name_label(var/new_label)` instead